### DISCUSSION TITLE
calendar view missing some months

### DISCUSSION THREAD

**Discussion by anarcat** [photoprism/photoprism#3214] [Q&A] [answered]
First, thank you for Photoprism. It looks like a great and powerful app, and it solves many problems in my workflow.

I have a question about the "Calendar" view in the Progressive Web App (PWA). It doesn't seem to show photos properly. For example, here's the top of the view (with some photos redacted, sorry):

![Screenshot 2023-02-20 at 11-37-57 PhotoPrism AI-Powered Photos App](https://user-images.githubusercontent.com/796623/220161447-128abe57-42f9-4034-9fef-863df86d6bfc.png)

Here you can see:

 * February 2023
 * January 2023
 * August 2022
 * July 2022
 * December 2021
 * November 2021
 * October 2021
 * September 2021
 * August 2021

Yet there *are* photos in 2022 other than just in July and August! Here, for example, is a screenshot of a photo from 2022-11-03:

![image](https://user-images.githubusercontent.com/796623/220162027-c5e7bb2d-f95e-4faf-9847-74d13efb5137.png)

That one is accessible in `/library/browse` (the Search section).  I have tried:

```
docker-compose -f /home/anarcat/compose-photoprism.yml exec photoprism photoprism index --force --cleanup 2022
```

to tell photoprism to rescan that year, to no avail. I'm now rescanning the entire library just in case that's the problem... 

Note that the Android app (MobilePrism) also shows the same behavior, which is more problematic (and also not your problem of course, but it shows there's something wrong in the "calendar")...

Also note that you can see a similar behavior in the demo app, in https://demo.photoprism.app/library/browse:

![image](https://user-images.githubusercontent.com/796623/220163554-a14fdb80-867d-455f-8cee-04880500e260.png)

Notice the second tile, from February 2023... Now look at the [calendar view](https://demo.photoprism.app/library/calendar), that video is missing:

![image](https://user-images.githubusercontent.com/796623/220163914-ffeffca8-b6c2-44c5-85cd-62dd99adbd04.png)

Any idea what could be happening here? Is this a bug I should report?

**Comment by graciousgrey** [comment]
The calendar is based on your photos metadata if months are missing it might be that some of your photos have wrong/missing metadata or the related photos are marked as private or archived. 

The demo behaves as expected as far as I can see, you can find the video in the respective calendar month: https://demo.photoprism.app/library/calendar/arqd73w3k298t5ms/view

**Reply by anarcat** [reply]
> The demo behaves as expected as far as I can see, you can find the video in the respective calendar month: https://demo.photoprism.app/library/calendar/arqd73w3k298t5ms/view

oh you're right about that part... :)

> The calendar is based on your photos metadata if months are missing it might be that some of your photos have wrong/missing metadata or the related photos are marked as private or archived.

hmm... i don't believe I have marked anything as private or archived here. and all photos *should* have proper metadata, and indeed if you look at the second screenshot, it's detected properly as well. just to  drill into that example again, here's the metadata of the photo that should show up in November 2022 but there's no November 2022 tile in teh calendar:

![image](https://user-images.githubusercontent.com/796623/220167720-c3188ef9-2e36-4ee3-af8e-0ee13fbfc93a.png)

Here you can see the `Taken` field correctly identifies it as November 2022... Yet there's no November 2022 tile in the calendar.

**Reply by graciousgrey** [reply]
Are you running the [latest PhotoPrism version](https://docs.photoprism.app/release-notes/) and what database do you use?

**Reply by graciousgrey** [reply]
Also make sure to wait until the initial indexing process has finished.

**Reply by anarcat** [reply]
i'm using the `photoprism:latest` docker image which is labeled as `Build 221118-e58fee0fb` here. and yeah, i'm reindexing everything so it will take a while for that to complete. should i be trying the `preview` image instead?

**Reply by graciousgrey** [reply]
Latest should be fine.

**Comment by anarcat** [comment]
just completed a `photoprism index --force --cleanup`, no change.

**Reply by graciousgrey** [reply]
Anything special about your setup? Settings that differ from the defaults? Are you using maridadb or sqlite? Did you modify the database manually in the past?

**Reply by anarcat** [reply]
i am using sqlite, did not touch the DB. here is my current docker compose file: https://gitlab.com/anarcat/puppet/-/blob/053f4caff184475c3db1786fc9bab7ba6dbb5eca/site-modules/profile/files/compose-photoprism.yml

**Reply by graciousgrey** [reply]
This looks good to me. I assume it has something to do with the files metadata. Could you send a few example files that do not appear in the calendar to samples@photoprism.app?

**Comment by anarcat** [comment, marked as answer]
ook, just about when i was packing up those file samples, the months magically appeared. i am not sure what changed: i didn't do anything since we last talked, but things seem to be resolved.

i don't know what the fix is, but this is fixed, I guess... weeeird! :)

**Reply by lastzero** [reply]
Glad it's working now!
